### PR TITLE
Support multiple keys per environment

### PR DESCRIPTION
**Pull request by erickduran**
- Supports new format for CERTIFICATE_AUTHORITIES where `name` is environment (and multiple keys can be added):
  ```
  [
  {
    'key': redacted,
    'crt': redacted,
    'name': 'staging',
    'passphrase': redacted, 
    'kid': 'staging_slot_a'
  }, ...
  ]
  ```
- Signing key must be activated for each environment as `ACTIVE_SIGNING_KEYS={"staging": "some_kid", "production": "some_kid"}`, otherwise endpoint will return 400. This will be the key used to sign tokens.
- Public JWKS will expose all keys per environment.

### CODE REVIEW

File: confidant/services/jwkmanager.py [hunk 1]
```diff
@@ -1,65 +1,87 @@
...
-                             passphrase=ca['passphrase'])
-
-    def set_key(self, kid: str, private_key: str,
+            for environment in CERTIFICATE_AUTHORITIES:
+                for ca in CERTIFICATE_AUTHORITIES[environment]:
+                    if validator.validate(ca):
+                        self.set_key(environment, ca['kid'], ca['key'],
+                                     passphrase=ca['passphrase'])
+                    else:
+                        logger.error(f'Invalid entry in {environment} '
```
Comment (skiptomyliu) [inline confidant/services/jwkmanager.py:44]: let's add the kid also.

File: confidant/settings.py [hunk 2]
```diff
@@ -636,3 +636,4 @@ def get(name, default=None):
 # Module that will perform an external ACL check on API endpoints
 ACL_MODULE = str_env('ACL_MODULE', 'confidant.authnz.rbac:default_acl')
 DEFAULT_JWT_EXPIRATION_SECONDS = int_env('DEFAULT_JWT_EXPIRATION_SECONDS', 3600)
+ACTIVE_SIGNING_KEYS = json.loads(str_env('ACTIVE_SIGNING_KEYS', '{}'))
```
Comment (skiptomyliu) [inline confidant/settings.py:639]: can we add a comment above on format expected here?

File: confidant/services/jwkmanager.py [hunk 3]
```diff
@@ -115,14 +137,24 @@ def get_payload(self, certificate: str, token: str,
...
+            return self._keys[environment].get_key(
+                ACTIVE_SIGNING_KEYS[environment]
+            )
+
+    def get_jwks(self, environment: str, algorithm: str = 'RS256') \
+            -> List[Dict[str, str]]:
+        keys = self._keys.get(environment)
+        if keys:
+            stats.incr(f'get_jwks.{environment}.hit')
+            return [{
```
Comment (skiptomyliu) [inline confidant/services/jwkmanager.py:151]: I think we want it in this format instead:


```
{
  "keys": [
    {
      "kty": "RSA",
      "e": "...",
      "n": "...",
      "alg": "RS256",
      "kid": "staging_1"
    },
{
      "kty": "RSA",
      "e": "...",
      "n": "...",
      "alg": "RS256",
      "kid": "staging_2"
    }
  ]
}
```

Comment (erickduran) [reply confidant/services/jwkmanager.py:151]: Yeah this is the final format (all fields are exported by `export_public()` except `alg`). The route itself wraps the keys in the `keys` object per the schema:
https://github.com/lyft/confidant/blob/master/confidant/schema/jwks.py#L23

**Review by skiptomyliu — APPROVED** [review]

File: confidant/settings.py [hunk 4]
```diff
@@ -606,6 +606,17 @@ def str_env(var_name, default=''):
 ENABLE_SAVE_LAST_DECRYPTION_TIME = bool_env('ENABLE_SAVE_LAST_DECRYPTION_TIME')
 
 # Add any certificate authorities
+# Should be in encrypted settings following this format (where name is
+# the name of the environment) and key ids must be unique:
+# [
```
Comment (skiptomyliu) [inline confidant/settings.py:611]: @erickduran just caught this now, I don't think this is the correct format anymore.

**Pull request merged**